surefire-api/src/main/java/org/apache/maven/surefire/api/report/LegacyPojoStackTraceWriter.java

```diff
@@ -54,7 +52,7 @@ public String writeTraceToString() {
             if (isMultiLineExceptionMessage(t)) {
                 // SUREFIRE-986
                 String exc = t.getClass().getName() + ": ";
-                if (StringUtils.startsWith(builder, exc)) {
+                if (builder.toString().startsWith(exc)) {
```

Comment:
I think you could use `builder.indexOf(exc) == 0` here to avoid creating the string

Reply:
Correct. That should be it.

Reply:
@FredrikAnderson

Reply:
Yes changed to proposed statement: if (builder.toString().startsWith( ...